transaction/core/src/ring_ct/rct_bulletproofs.rs

```diff
@@ -492,13 +610,15 @@ fn sign_with_balance_check<CSPRNG: RngCore + CryptoRng, S: RingSigner + ?Sized>(
                     return Err(Error::SignedInputRulesNotAllowed);
                 }
                 InputRing::Signable(_) => {}
+                InputRing::ViewOnly(_) => {}
             };
         }
     }
 
     if rings.is_empty() {
         return Err(Error::NoInputs);
     }
+
```

Comment:
Please look at line 625 below. Looks like the code that calculates `ring_size` will fail if all input rings are ViewOnly

Reply:
We no longer have ViewOnly but it might be worth changing the `if` below to a match statement, so that if we ever add a variant this will be obvious it needs addressing here.

Reply:
:+1: that seems like a good idea, thank you